tests/test_enforce.py

```diff
@@ -95,3 +103,62 @@ def nonfinal_classmethod(cls):
 
         self.assertNotEqual(ClassMethodOverrider.nonfinal_classmethod(),
                             Enforcing.nonfinal_classmethod())
+
+    def test_is_compatible_when_compatible(self):
+        def foo(x, /, y: str, *args, z: int, **kwargs) -> object:
+            pass
+
+        def bar(x, /, y: object, a: float, *, b: bytes, z: int, **kwargs) -> str:
+            pass
+
+        is_compatible(foo, bar)
+
+    def test_is_compatible_when_return_types_are_incompatible(self):
+        def foo(x) -> int:
+            pass
+
+        def bar(x) -> str:
+            pass
+
+        with self.assertRaises(TypeError):
+            is_compatible(foo, bar)
```

Comment:
The name of the method could be ensure_compatible or something like that. is_compatible in my opinion suggests a boolean return value.